```
Failure applying sagemath/sage#41944 as a patch, resetting
fatal: Resolve operation not in progress, we are not resuming.
ERROR [transfer|run:137]: HTTP Error 503: Service Unavailable
************************************************************************
Traceback (most recent call last):
  File "/home/runner/work/sage/sage/build/bin/../sage_bootstrap/download/cmdline.py", line 127, in run_safe
    run()
  File "/home/runner/work/sage/sage/build/bin/../sage_bootstrap/download/cmdline.py", line 107, in run
    app.download_url(args.url_or_tarball, args.destination)
  File "/home/runner/work/sage/sage/build/bin/../sage_bootstrap/download/app.py", line 37, in download_url
    Download(url, destination, progress=not self.quiet, ignore_errors=False).run()
  File "/home/runner/work/sage/sage/build/bin/../sage_bootstrap/download/transfer.py", line 139, in run
    raise error
  File "/home/runner/work/sage/sage/build/bin/../sage_bootstrap/download/transfer.py", line 133, in run
    urlretrieve(self.url, self.destination)
  File "/usr/lib/python3.12/urllib/request.py", line 240, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
                            ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 215, in urlopen
    return opener.open(url, data, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 521, in open
    response = meth(req, response)
               ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 630, in http_response
    response = self.parent.error(
               ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 553, in error
    result = self._call_chain(*args)
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 492, in _call_chain
    result = func(*args)
             ^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 745, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 521, in open
    response = meth(req, response)
               ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 630, in http_response
    response = self.parent.error(
               ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 559, in error
    return self._call_chain(*args)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 492, in _call_chain
    result = func(*args)
             ^^^^^^^^^^^
  File "/usr/lib/python3.12/urllib/request.py", line 639, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 503: Service Unavailable
************************************************************************
```
Sometimes flaky error due to network. Now this happens usually.

--- a/.github/workflows/merge-fixes.sh
+++ b/.github/workflows/merge-fixes.sh
@@ -47,7 +47,20 @@ for REPO in ${SAGE_CI_FIXES_FROM_REPOSITORIES:-sagemath/sage}; do
                     PULL_URL="https://github.com/$REPO/pull/$a"
                     PULL_SHORT="$REPO#$a"
                     PULL_FILE="$REPO_FILE-$a"
-                    PATH=build/bin:$PATH build/bin/sage-download-file --quiet "$PULL_URL.patch" $PULL_FILE.patch
+                    DOWNLOAD_SUCCESS=
+                    for RETRY in 1 2 3; do
+                        if PATH=build/bin:$PATH build/bin/sage-download-file --quiet "$PULL_URL.patch" $PULL_FILE.patch; then
+                            DOWNLOAD_SUCCESS=yes
+                            break
+                        else
+                            echo "Download attempt $RETRY/3 failed for $PULL_URL.patch"
+                            sleep $((RETRY * 5))
+                        fi
+                    done
+                    if [ -z "$DOWNLOAD_SUCCESS" ]; then
+                        echo "Failed to download $PULL_URL.patch after 3 attempts, skipping"
+                        continue
+                    fi
                     date -u +"%Y-%m-%dT%H:%M:%SZ" > $PULL_FILE.date  # Record the date, for future reference
                     LAST_SHA=$(sed -n -E '/^From [0-9a-f]{40}/s/^From ([0-9a-f]{40}).*/\1/p' $PULL_FILE.patch | tail -n 1)
                     COMMITS_URL="https://github.com/$REPO/commits/$LAST_SHA"

